plugin.video.youtube/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_player.py

```diff
@@ -14,13 +22,16 @@ def __init__(self, player_type, context):
 
         self._context = context
 
-    def play(self, playlist_index=-1):
+    def play(self, playlist_index=-1, list_item=None):
         """
         We call the player in this way, because 'Player.play(...)' will call the addon again while the instance is
         running.  This is somehow shitty, because we couldn't release any resources and in our case we couldn't release
         the cache. So this is the solution to prevent a locked database (sqlite).
         """
-        self._context.execute('Playlist.PlayOffset(%s,%d)' % (self._player_type, playlist_index))
+        if list_item is None:
+            self._context.execute('Playlist.PlayOffset(%s,%d)' % (self._player_type, playlist_index))
+        else:
+            xbmc.Player().play(item=list_item.getPath(), listitem=list_item)
```

Comment:
Any reason for calling Player.play() directly?

Reply:
This was left over, removed now